### PR TITLE
Various default value fixes - consolidated @AtkinsChang and @trevorsibanda with minor additions

### PR DESCRIPTION
**Pull request by rbellamy**
Replaces PR's #1551, #1582 and #1583.

Merged and resolved conflicts:
@AtkinsChang branch pg_profile - for proper handling of Postgres default values
@trevorsibanda branches fix-codegen-decimaltype-default-0.00 and fix-issue-1557 - proper handling of BigDecimal defaults within JdbcModelBuilder and a MySQL test.

Some minor changes on my part:

Trim Oracle BigDecimal default value to remove trailing space - otherwise NumberFormatException
Rearranged MySQL tests.
Added Oracle test for BigDecimal default values.
Passes all tests for H2, Oracle, Postgres and MySQL except testDiscriminatorTest, which fails for MySQL. However, that test is failing in master as well #1608.

My justification for taking the approach of pulling several branches from other folks is that taken in a piecemeal fashion, each PR was incomplete or inconsistent, or one required the other - e.g. to get full support of Postgres defaults, the branch from @AtkinsChang was needed, in addition to the changes to JdbcModelBuilder found in one of @trevorsibanda branches, or a change in @trevorsibanda MySQL fix branch altered the default value handling in the MySQLProfile rather than where it was best managed, in the JdbcModelBuilder.

Maybe this will be perceived as me forcing myself on a process that doesn't need it - my hope is that this PR will be considered in the intent in which I make it - as a genuine effort to help.

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @rbellamy,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by rbellamy** [comment]
Would someone please explain to me why the `typesafe-cla-validator` is failing the check?
selection 1043x793 20160912100654

File: slick-testkit/src/codegen/resources/dbs/postgres.sql [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                       "text_default_null" TEXT DEFAULT NULL,
+                       -- for String
+                       "char_default_empty" CHAR(2) DEFAULT 'ab',
+                       "char_default_null" CHAR(2) DEFAULT NULL,
+                       -- for Char
+                       "char1_default_empty" CHAR(1) DEFAULT '',
+                       "char1_default_valid" CHAR(1) DEFAULT 'a',
+                       "char1_default_invalid" CHAR(1) DEFAULT 'ab',
+                       "char1_default_null" CHAR(1) DEFAULT NULL
+                      );
```
Comment (trevorsibanda) [inline slick-testkit/src/codegen/resources/dbs/postgres.sql]: Newline here

**Comment by typesafehub-validator** [comment]
Hi @rbellamy,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by rbellamy** [comment]
I'm completely confused by the `typesafe-cla-validator` - it keeps complaining that I haven't signed, but I've done - at this point about a half a dozen times. Could someone from Lightbend help me out here? I'm not sure what else I need to do...

**Comment by cvogt** [comment]
@szeiger could you look into the cla validator problems? It does indeed not recognize anything having been signed.

**Comment by d6y** [comment]
There's another PR with a cla validator problem in https://github.com/slick/slick/pull/1613 /cc @szeiger

**Comment by rbellamy** [comment]
@cvogt and @szeiger - any feedback on a timeline for getting the check resolved?

**Comment by Gacko** [comment]
It works now! :)

**Comment by rbellamy** [comment]
@Gacko - did you close your PR and open a new one? If not, how did you trigger a re-check?

**Comment by rbellamy** [comment]
Okay, I'm going to close this PR and open a new one to see if the check passes this time.

**Comment by Gacko** [comment]
I just clicked the link in @typesafehub-validator's checklist.

**Comment by cvogt** [comment]
validator should be fine now @rbellamy

**Pull request closed without merging**